### PR TITLE
fix: reschedule or cancel

### PR DESCRIPTION
**Pull request by Chiranjeev-droid**
## What does this PR do?

This pr add an option to make the cancelation reason required / optional within event types.

- Fixes #15977 

 Loom Video: https://www.loom.com/share/4c0422dd659345899d936d23ab20c7e0?sid=6bcfa671-8bcd-48fb-b983-d70c2ab01d7f

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [x] I have added a Docs issue [here](https://github.com/calcom/docs/issues/new) if this PR makes changes that would require a [documentation change](https://docs.cal.com/docs). If N/A, write N/A here and check the checkbox.
- [x] I confirm automated tests are in place that prove my fix is effective or that my feature works.

## How should this be tested?
Go to advanced tab of event types and make the cancellation reason field optional / required and change the default label or placeholder if needed for the  cancel reason field.
Then create a booking and try cancelling it , you will then be able to see the changes.

### CODE REVIEW

File: apps/web/public/static/locales/cs/common.json [hunk 1]
```diff
@@ -46,6 +46,8 @@
   "need_to_reschedule_or_cancel": "Přeplánovat nebo zrušit?",
   "no_options_available": "Nejsou k dispozici žádné možnosti",
   "cancellation_reason": "Důvod zrušení",
+  "cancellation_reason_required": "Důvod zrušení (Povinný)",
+  "cancellation_reason_optional": "Důvod zrušení (Volitelný)",
```
Comment (anikdhabal) [inline apps/web/public/static/locales/cs/common.json:50]: You don't need to add this in every locales. Only add in English

File: apps/web/components/booking/CancelBooking.tsx [hunk 2]
```diff
@@ -31,7 +32,7 @@ type Props = {
       email: string;
       timeZone?: string;
     };
-    eventType: unknown;
+    eventType: any;
```
Comment (anikdhabal) [inline apps/web/components/booking/CancelBooking.tsx:35]: This is not a good practice. You can get the eventType from `RouterOutput`. Please refer to another component where the eventType type is defined

File: packages/features/form-builder/FormBuilder.tsx [hunk 3]
```diff
@@ -531,10 +531,13 @@ function FieldEditDialog({
                       }
                       label={t("identifier")}
                     />
-                    <CheckboxField
-                      description={t("disable_input_if_prefilled")}
-                      {...fieldForm.register("disableOnPrefill", { setValueAs: Boolean })}
-                    />
+
+                    {fieldForm.getValues("name") !== "cancelReason" && (
+                      <CheckboxField
```
Comment (anikdhabal) [inline packages/features/form-builder/FormBuilder.tsx:536]: Why shouldn't the prefill input checkbox be visible for the cancellation reason?

File: apps/web/components/booking/CancelBooking.tsx (hunk 2 above)
Comment (Chiranjeev-droid) [reply apps/web/components/booking/CancelBooking.tsx:35]: ok , thanks for reviewing ,will make the changes and reopen it.

Comment (anikdhabal) [reply apps/web/components/booking/CancelBooking.tsx:35]: you can'r reopen it, so tag me here after finishing the work

**Pull request closed without merging**